File: two-pointers/easy-0125-valid-palindrome/Solution.java
class Solution {
    public boolean isPalindrome(String s) {
        int i = 0;
        int j = s.length() - 1;

        while (i < j) {
            while (i < s.length() && !Character.isLetterOrDigit(s.charAt(i))) i++;
            while (j > -1 && !Character.isLetterOrDigit(s.charAt(j))) j--;

            if (i > j) break;

            if (Character.toLowerCase(s.charAt(i)) != Character.toLowerCase(s.charAt(j))) return false;
            i++;
            j--;
        }

        return true;
    }
}

File: two-pointers/easy-0125-valid-palindrome/Solution.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        n = len(s)
        i = 0
        j = n - 1

        while (i < j):
            while (i < n  and not s[i].isalnum()): i += 1
            while (j > -1 and not s[j].isalnum()): j -= 1

            if (i >= j): break

            if (s[i] != s[j]): 
                print(i, j, s[i], s[j])
                return False
            else:
                i += 1
                j -= 1

        return True